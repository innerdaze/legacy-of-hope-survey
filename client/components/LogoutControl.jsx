import React, { Component } from 'react'
import Menu from 'grommet/components/Menu'
import Anchor from 'grommet/components/Anchor'

export default ({ handleLogoutPress = Function.prototype }) => (
  <Menu primary={true}>
    <Anchor
      label='Logout'
      onClick={handleLogoutPress}
      />
  </Menu>
)
