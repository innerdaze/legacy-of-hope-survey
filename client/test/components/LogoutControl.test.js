import React from 'react'
import LogoutControl from '../../components/LogoutControl'

let wrapper, handleLogoutPress

describe('Testing on LogoutControl...', () => {
  beforeEach(() => {
    handleLogoutPress = jest.fn()
    wrapper = shallow(<LogoutControl handleLogoutPress={handleLogoutPress} />)
  })

  describe('LogoutControl layout', () => {
    test('Expect LogoutControl to have Menu parent comp and a Anchor child comp', () => {
      expect(wrapper.find('Menu')).toHaveLength(1)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on prop', () => {
    test('Expect handleLogoutPress to be called if Anchor click', () => {
      wrapper.find('Anchor').simulate('click')
      expect(handleLogoutPress).toBeCalled()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
