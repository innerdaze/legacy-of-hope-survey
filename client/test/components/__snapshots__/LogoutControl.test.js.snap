// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on LogoutControl... LogoutControl layout Expect LogoutControl to have Menu parent comp and a Anchor child comp 1`] = `
<Menu
  closeOnClick={true}
  direction="column"
  dropAlign={
    Object {
      "left": "left",
      "top": "top",
    }
  }
  pad="none"
  primary={true}
>
  <Anchor
    animateIcon={true}
    label="Logout"
    method="push"
    onClick={[MockFunction]}
    tag="a"
  />
</Menu>
`;

exports[`Testing on LogoutControl... Test on prop Expect handleLogoutPress to be called if Anchor click 1`] = `
<Menu
  closeOnClick={true}
  direction="column"
  dropAlign={
    Object {
      "left": "left",
      "top": "top",
    }
  }
  pad="none"
  primary={true}
>
  <Anchor
    animateIcon={true}
    label="Logout"
    method="push"
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    tag="a"
  />
</Menu>
`;
